src/game_battler.cpp

```diff
@@ -456,10 +456,11 @@ void Game_Battler::RemoveBattleStates() {
 	if (IsDead() &&
 		non_permanent(1)) {
 		ChangeHp(1);
+		states[1] = 0;
```

Comment:
When using RemoveState or ReaderUtil::GetElement(), the index starts at 1. When using states[] directly in an array, the index starts at 0. This means in this case you are disabling the second state (that isn't Dead), not the first one. You can solve this by either:
states[0] = 0;
RemoveState(1);

Reply:
Fixed, thanks!